app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -384,6 +385,38 @@ def openshift_disk_space
     end
   end
 
+  def add_rhv_engine
+    engine_id = params['engine_id']
+    engine_name = params['engine_name']
+
+    # don't delete/add rows if nothing changed
+    if (@deployment.rhev_engine_host_id != engine_id) ||
+       (@deployment.rhev_self_hosted_engine_hostname != engine_name)
+
+      rhv_engine = DeploymentHost.where(deployment_id: @deployment.id,
+                           discovered_host_id: @deployment.rhev_engine_host_id,
```

Comment:
searching for a discovered_host_id: nil always yielded no rhv engine.  Omitting discovered_host_id fixes it:

```ruby
rhv_engine = DeploymentHost.where(
  deployment_id: @deployment.id, 
  deployment_host_type: 'rhv_engine'
).first
```